/kf_tracker_node:
  ros__parameters:
    # Rate of prediction step
    dt_pred: 0.02
    print_debug_msg: False

    # Diagonal elements of the process covariance matrix Q, for constant acceleration model
    # [x-x, y-y, z-z, vx-vx, vy-vy, vz-vz, ax-ax, ay-ay, az-az]
    # NOT USED
    q_diag: [0.1, 0.1, 0.1, 0.1, 0.1, 0.1 ,0.1 , 0.1, 0.1]

    # Diagonal elements of the process covariance matrix Q, for constant velocity model
    # # [x-x, y-y, z-z, vx-vx, vy-vy, vz-vz]
    # NOT USED
    # q_diag: [5,5,5,10,10,10]

    # Diagonal elements of the process covariance matrix Q, for constant velocity or acceleration models
    # Also used in initializing the upper left (position) part of the P matrix (process covariance) for a track
    # This is a variance in m^2 (squre root of standard deviation in m)
    r_diag: [0.01, 0.01, 0.01]

    # Standard deviation of the random acceleration variable.
    # Used in computing Q matrix
    # reasonable value, such as between 1.0 and 5.0 m/sÂ², depending on the expected maximum acceleration of the objects you're tracking
    sigma_a: 10.0

    # Standard deviation of position
    # Adjust sigma_p_ and sigma_v_ based on how much you trust your initial position and velocity estimates.
    # If you have high confidence in the initial positions but less in velocities, you might set sigma_p_ to a smaller value (e.g., 0.5) 
    # and sigma_v_ to a larger value (e.g., 2.0).
    
    # Used in computing initial P (state estimate covariance) matrix
    sigma_p: 0.5
    # Standard deviation of velocity
    # Used in computing initial P (state estimate covariance) matrix
    sigma_v: 2.0

    # Standard deviation of turn rate in rad/sec
    sigma_omega: 0.1
    
    # Standard deviation of heading angle in radians
    sigma_theta: 0.2

    #Standard deviation of climb angle in radians*
    sigma_gamma: 0.15


    # Provide estimate w.r.t this frame
    tracking_frame: 'observer/odom'
    # Topic name of the tag detections
    apriltags_topic: '/tag_detections'
    # Frame name of the detected target
    target_frameid: 'tag'

    # Get target measurements from TF tree
    listen_tf: False

    # Maximum uncertainty before rejecting a track [m^3] 
    V_max: 100.0
    # Minimum uncertainty for confirming a track [m^3]
    V_certain: 30.0
    # Minimum number of measurements to confirm a track
    N_meas: 10
    # Measurement association log-likelihood threshold
    # A very low threshold like -100 may result in associating unlikely measurements with tracks.
    l_threshold: -10.0

    # Maximum distance between  a state & measurement to consider them as a match (currently not used)
    dist_threshold: 5.0

    # False: Doesn't consider measurement track ID in measurement-state association 
    use_track_id: false

    # Maximum time (seconds) from last measurement before considering a track uncertain and removing it
    track_mesurement_timeout: 1.5
    #use_sim_time : true
